File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/max-sum-of-a-pair-with-equal-sum-of-digits.cpp
class Solution {
public:
    int maximumSum(vector<int>& nums) {
        unordered_map<int,vector<long long>> m;
        for(int i=0; i<nums.size(); i++){
            int x = nums[i];
            int sum = 0;
            while(x > 0){
                sum += x%10;
                x/=10;
            }
            m[sum].push_back(nums[i]);
        }

        for(auto &it:m){
            sort(it.second.rbegin(), it.second.rend());
        }
        long long ans = -1;
        for(auto &it:m){
            vector<long long> x = it.second;
            if (x.size() > 1)
                ans = max(ans, x[0] + x[1]);
        }
        return ans;
    }
};

File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/max-sum-of-a-pair-with-equal-sum-of-digits.go
func maximumSum(nums []int) int {
    var sum_digits [82]int
    max_sum := -1
    var sum, n int
    for i := range nums {
        for sum, n = 0, nums[i]; n > 0; n/=10 {
            sum += (n%10)
        }

        if sum_digits[sum] == 0 {
            sum_digits[sum] = nums[i]
        } else {
            max_sum = max(max_sum, (sum_digits[sum] + nums[i]))

            if nums[i] > sum_digits[sum] {
                sum_digits[sum] = nums[i]
            }
        }
    }

    return max_sum
}